### PR TITLE
JP-3734: Making OLS_C Default for CI Tests

### PR DESCRIPTION
**Pull request by kmacdonald-stsci**
…ith CHARGELOSS.

Resolves [JP-3734](https://jira.stsci.edu/browse/JP-3734)

Closes #

The JWST default algorithm was changed to be OLS_C for ramp fitting.  The STCAL ramp fit testing has been changed to reflect this change in default algorithm.

## Tasks
- [ ] update or add relevant tests
- [ ] update relevant docstrings and / or `docs/` page
- [ ] Does this PR change any API used downstream? (if not, label with `no-changelog-entry-needed`)
  - [ ] write news fragment(s) in `changes/`: `echo "changed something" > changes/<PR#>.<changetype>.rst` (see below for change types)
  - [ ] run regression tests with this branch installed (`"git+https://github.com/<fork>/stcal@<branch>"`)
    - [ ] [`jwst` regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/jwst.yml) 
    - [ ] [`romancal` regression test](https://github.com/spacetelescope/RegressionTests/actions/workflows/romancal.yml)

<details><summary>news fragment change types...</summary>

- ``changes/<PR#>.apichange.rst``: change to public API
- ``changes/<PR#>.bugfix.rst``: fixes an issue
- ``changes/<PR#>.general.rst``: infrastructure or miscellaneous change
</details>

### CODE REVIEW

File: src/stcal/ramp_fitting/ramp_fit_class.py [hunk 1]
```diff
@@ -47,8 +47,8 @@ def __init__(self):
         self.suppress_one_group_ramps = False
 
         # C code debugging switch.
-        self.run_c_code = False
-        self.run_chargeloss = True
+        self.run_c_code = True
+        self.run_chargeloss = False
         # self.run_chargeloss = False
```
Comment (melanieclarke) [inline src/stcal/ramp_fitting/ramp_fit_class.py:52]: Remove commented out code

**Review by melanieclarke — COMMENTED** [review]
Looks fine to me!  One very small clean up request.

Did you run regression tests somewhere?

**Comment by melanieclarke** [comment]
@kmacdonald-stsci - did regression tests get run for JWST for this change?

**Comment by tapastro** [comment]
It looks like this is breaking a unit test:

`FAILED tests/test_ramp_fitting.py::test_neg_med_rates_single_integration_multi_segment_optional - AssertionError: 
  Not equal to tolerance rtol=1e-06, atol=0
  
  -inf location mismatch:
   ACTUAL: array([ 0.     , 60.53071,     -inf], dtype=float32)
   DESIRED: array([0., 0., 0.])
 `
 
 We should probably resolve this failure before merging.

**Comment by kmacdonald-stsci** [comment]
> It looks like this is breaking a unit test:
> 
> ```
> FAILED tests/test_ramp_fitting.py::test_neg_med_rates_single_integration_multi_segment_optional - AssertionError: 
>   Not equal to tolerance rtol=1e-06, atol=0
>   
>   -inf location mismatch:
>    ACTUAL: array([ 0.     , 60.53071,     -inf], dtype=float32)
>    DESIRED: array([0., 0., 0.])
> ```
> 
> We should probably resolve this failure before merging.

It's not clear to me this is caused by this PR.  And I've never seen such a strange problem like this.  If I run `pytest tests/test_ramp_fitting.py` the output to be checked for this test is different that if I run `pytest tests/test_ramp_fitting.py::test_neg_med_rates_single_integration_multi_segment_optional`.  

It doesn't matter if I run with the `OLS` algorithm or the `OLS_C` algorithm.  The screenshots of the test outputs show that the returned array is all zero when running the whole test script, but is non-zero if only the specific test is run, regardless of the choice of algorithm.  Because this difference occurs independent of which algorithm is chosen, I don't think this PR can be the problem.

I'll continue to investigate.

 


Screenshot 2024-10-03 at 3 08 59 PM
Screenshot 2024-10-03 at 3 20 23 PM

**Pull request closed without merging**